interop-testing/src/main/java/io/grpc/testing/integration/AbstractInteropTest.java

```diff
@@ -415,6 +415,42 @@ protected final ServerStreamTracer.Factory createCustomCensusTracerFactory() {
         true, true, true, false /* real-time metrics */);
   }
 
+  /* Parses input string as a semi-colon-separated list of colon-separated key/value pairs.
+   * Allow any character but semicolons in values.
+   * If the string is emtpy, return null.
+   * Otherwise, return a client interceptor which inserts the provided metadata.
+   */
+  @Nullable
+  protected final ClientInterceptor maybeCreateAdditionalMetadataInterceptor(
+      String additionalMd)
+      throws IllegalArgumentException {
+    if (additionalMd.length() == 0) {
+      return null;
+    }
+    Metadata m = new Metadata();
+    while (additionalMd.length() > 0) {
+      int i = additionalMd.indexOf(':');
```

Comment:
Although simpler, suggested approach doesn't allow semi-colons in values.

Per the flag's spec, values are supposed to allow any ascii except colons.

Probably not the end of the world to disallow semi-colons too, but C++ and Go are allowing them so I think it's better as is for consistency.

Reply:
I didn't realize it was like that. That's an incredibly complicated format. I thuoght only semicolons wasn't allowed. If colons aren't allowed, we should have done colon-separated with equals between k/v.

Reply:
What do you think about updating the doc to disallow colons in values as well? Then, we can take your suggestion here and keep things simple.

Problem with changing separator format to `:` and `=` now is it will add more delay to RLS qualification.

Reply:
FWIW too, note that the format was not decided by me :), https://github.com/grpc/grpc/pull/18156/files did it this way a while back

Reply:
Updated per your original suggestion, https://github.com/grpc/grpc/pull/33253 to update the doc

Reply:
I have a bit more time to look into this now.

> Per the flag's spec, values are supposed to allow any ascii except colons.

That's not what it said. It said "Values may contain any character except semi-colons." So you can do the code as I mentioned. Was the text wrong in interop-test-descriptions.md?

Reply:
Sorry my statement "values are supposed to allow any ascii except colons." was wrong. The doc was right.

I should have said, "values are supposed to allow any ascii except semi-colons, i.e. colons should be allowed".

I went ahead and updated to the code you mentioned in any case.

Reply:
Note that `"abc-key:abc:value".split(":", 2)` returns `["abc-key", "abc:value"]`.

Reply:
Whoops, missed that! I can close https://github.com/grpc/grpc/pull/33253 then.